### ISSUE TITLE
[speech-to-text] Add additional callback routines to SpeechToText delegate

### ISSUE THREAD

**Issue by rfdickerson** [watson-developer-cloud/ios-sdk#152]
Currently the callback routines in the delegate is limited only to:

```swift
func onSpeechRecognized(text: String) 
```

For some use cases, clients will want to see the decibels on the signal, and perhaps when onSpeechBegin() and onSpeechEnd().

**Comment by dsato80** [comment]
+1

I want to see decibels on the signal. 

One concern to get power is that `AudioQueueGetProperty` with `kAudioQueueProperty_CurrentLevelMeterDB` is not good for showing mic indicator while speaking because the window size for power calculation is too longer (about 1,000 msec) to show quick responsive indicator. 

Indicator with the power provided by the system will gradually decreasing in one second after you stop speaking. It is not good user interface based on our experience.

So there are two solutions
1. implement customisable power calculator in the lib
2. expose callback with raw audio data (with customisable audio buffer length for `AudioQueueAllocateBuffer`)

**Comment by dsato80** [comment]
want to get SpeechToTextResult 

```objective-c
func onSpeechRecognized(text: String, result:SpeechToTextResult) 
```

**Comment by glennrfisher** [comment]
Closed in favor of #168.
